crates/typos-cli/Cargo.toml

```diff
@@ -77,7 +77,7 @@ colorchoice-clap = "1.0.3"
 serde_regex = "1.1.0"
 regex = "1.10.4"
 encoding_rs = "0.8.34"
-serde-sarif = "0.4.2"
+serde-sarif = "0.5.0"
```

Comment:
Why was this upgrade done?  Why wasn't it in the prior commit?

Reply:
Please take a loot at https://github.com/psastras/sarif-rs/issues/646.

Reply:
That doesn't answer either question on its own.

Reply:
This version makes no code changes, bumps the version number, updates some dependencies, and then removes the experimental warning.
The bumping of the version was done to address the issue of https://github.com/crate-ci/typos/pull/1047#discussion_r1659499185

Reply:
Or in other words...

> Why was this upgrade done? 

To pull in a version that was listed as unstable. This had no other affect as any breaking changes in that release do not affect us.

>  Why wasn't it in the prior commit?

*This part is still left unanswered*

Reply:
At the time I wrote the initial code for this PR, that version had not yet been released. Since you mentioned that the dependency was unstable, I went and asked the author, who bumped up the version number, and I followed the update as a result.

Reply:
So this is an update for the reviewer and has no distinct meaning as a separate commit?  Then that should be squashed like the other cases that have been squashed before.

Reply:
squashed

Reply:
ok